roles/platform/tasks/setup_azure_authz.yml

```diff
@@ -108,128 +108,107 @@
         secret: "{{ __azure_xaccount_app_pword }}"
 
 - name: Request creation of Azure Managed Identities
-  when: ( __azure_identity_list_names is undefined ) or ( __azure_msi_item not in __azure_identity_list_names )
   loop_control:
     loop_var: __azure_msi_item
   register: __azure_az_msi_info
-  command: "az identity create -g {{ plat__azure_metagroup_name }} -n {{ __azure_msi_item }}"
+  azure.azcollection.azure_rm_resource:
+    resource_group: "{{ plat__azure_metagroup_name }}"
+    provider: ManagedIdentity
+    resource_type: userAssignedIdentities
+    resource_name: "{{ __azure_msi_item }}"
+    api_version: '2018-11-30'
+    idempotency: yes
+    state: present
+    body:
+      location: "{{ plat__region }}"
   loop:
     - "{{ plat__azure_idbroker_identity_name }}"
     - "{{ plat__azure_datalakeadmin_identity_name }}"
     - "{{ plat__azure_log_identity_name }}"
     - "{{ plat__azure_ranger_audit_identity_name }}"
 
-- name: Wait for identities to be listed
-  command: "az identity list -g {{ plat__azure_metagroup_name }}"
+- name: Wait for MSIs to be listed
+  azure.azcollection.azure_rm_resource_info:
+    resource_group: "{{ plat__azure_metagroup_name }}"
+    provider: ManagedIdentity
+    resource_type: userAssignedIdentities
+    api_version: '2018-11-30'
   register: __azure_identity_list
-  failed_when: __azure_identity_list.rc != 0
   delay: 5
   retries: 10
```

Comment:
Might be worth updating either the delay or retries. Sometimes this can take up to 5-10 minutes. As long as this bails out as soon as it discovers them (which my test run validated) we should up it. It doesn't make me feel good. But it is what it is.

Reply:
Upped the retries to 10 mins